src/generateSdk/astGenerator/GoAstGenerator.ts

```diff
@@ -0,0 +1,144 @@
+
+        // Check if the go ast generator is compiled
+        const goAstGeneratorPath = path.join(os.homedir(), ".golang_ast_generator", binaryName);
+        if (!fs.existsSync(goAstGeneratorPath)) {
+            await this.#compileGenezioGoAstExtractor();
+        }
+        const classAbsolutePath = path.resolve(input.class.path);
+        const result = await runNewProcessWithResultAndReturnCode(
+            `${goAstGeneratorPath} ${classAbsolutePath}`,
+        );
+        if (result.code !== 0) {
+            console.error(result.stderr);
+            throw new Error("Error: Failed to generate AST for class " + input.class.path);
+        }
+
+        const ast = JSON.parse(result.stdout);
+        const error = ast.error;
+        if (error) {
+            log.error(error);
+            throw new Error("Error: Failed to generate AST for class " + input.class.path);
+        }
+        const goAstBody = ast.body;
+        if (!goAstBody) {
+            throw new Error("Error: Failed to generate AST for class " + input.class.path);
+        }
+
+        const body: Node[] = [];
+
+        for (const astNode of goAstBody) {
+            if (astNode.type == AstNodeType.ClassDefinition) {
+                const classDefinition: ClassDefinition = {
+                    type: AstNodeType.ClassDefinition,
+                    name: astNode.name,
+                    path: astNode.path,
+                    methods: [],
+                };
+                for (const method of astNode.methods) {
+                    const methodDefinition: MethodDefinition = {
+                        type: AstNodeType.MethodDefinition,
+                        name: method.name,
+                        params: [],
+                        kind: MethodKindEnum.method,
+                        static: false,
+                        returnType: method.returnType,
+                    };
+                    for (const param of method.params) {
+                        methodDefinition.params.push({
+                            type: AstNodeType.ParameterDefinition,
+                            name: param.name,
+                            optional: param.optional,
+                            rawType: param.rawType,
+                            paramType: param.paramType,
+                        });
+                    }
+                    classDefinition.methods.push(methodDefinition);
+                }
+                body.push(classDefinition);
+            } else {
+                switch (astNode.type) {
+                    case AstNodeType.StructLiteral:
+                        const structLiteral: StructLiteral = {
+                            type: AstNodeType.StructLiteral,
+                            name: astNode.name,
+                            path: astNode.path,
+                            typeLiteral: astNode.typeLiteral,
+                        };
+                        body.push(structLiteral);
+                        break;
```

Comment:
Unexpected lexical declaration in case block. - eslint (no-case-declarations)

```suggestion
                    case AstNodeType.StructLiteral: {
                        const structLiteral: StructLiteral = {
                            type: AstNodeType.StructLiteral,
                            name: astNode.name,
                            path: astNode.path,
                            typeLiteral: astNode.typeLiteral,
                        };
                        body.push(structLiteral);
                        break;
                    }
```